src/main/scala/nl.knaw.dans.easy.deposit/package.scala

```diff
@@ -71,4 +74,19 @@ package object deposit {
         .getOrElse(onSuccess)
     }
   }
+
+  @throws[IOException]("when files can not be moved atomically from src to target")
+  def sameMounts(srcProvider: FileSystemProvider, src: File, target: File): Unit = {
+    val fileName = "same-mount-check"
+    val tempFile = src / fileName
+    val movedTempFile = target / fileName
+    Try {
+      if (tempFile.notExists) tempFile.createFile()
+      srcProvider.move(tempFile.path, movedTempFile.path, StandardCopyOption.ATOMIC_MOVE)
+      movedTempFile.delete()
+    }.recoverWith { case e =>
+      tempFile.delete()
```

Comment:
`tempFile.delete()` is just a side-effect. Use `doIfFailure` instead.

Reply:
It is called from constructors so I need to throw. `doIfFailure` returns a try.